### PR TITLE
docstore: queries should always return the key and revision fields

### PR DESCRIPTION
**Pull request by jba**

### CODE REVIEW

File: internal/docstore/driver/util.go [hunk 1]
```diff
@@ -160,3 +160,8 @@ func fpEqual(fp1, fp2 []string) bool {
 	}
 	return true
 }
+
+// FieldPathEqualsField reports whether a field path equals a field.
+func FieldPathEqualsField(fp []string, s string) bool {
```
Comment (shantuo) [inline internal/docstore/driver/util.go]: Why do we not care `fp` and `s` have multiple elements?

Comment (jba) [reply internal/docstore/driver/util.go]: This case comes up a lot: we want to compare a field path to a single field (basically a field path with exactly one element). For example, the revision field has to be a single field.

Comment (jba) [reply internal/docstore/driver/util.go]: PTAL

Comment (jba) [reply internal/docstore/driver/util.go]: Note that this function is just a convenience for
```
FieldPathsEqual(fp, []string{s})
```

Comment (shantuo) [reply internal/docstore/driver/util.go]: I see, please add a comment to not be used wrong.

Comment (jba) [reply internal/docstore/driver/util.go]: in next PR

**Pull request merged**